kong/cache.lua

```diff
@@ -112,8 +112,12 @@ local mt = { __index = _M }
 
 
 function _M.new(opts)
-  if _init then
-    error("kong.cache was already created", 2)
+  if not opts.shm_name or type(opts.shm_name) ~= "string" then
```

Comment:
AFAICT, this is equivalent to just `if type(opts.shm_name) ~= 'string' then...`.  also note that there's no performance advantage in checking the truthyness value first, the type-specific nature of compiled traces allow type checks to be compiled away into a simple constant (but that's only relevant in the tightest loops).

Reply:
Yes. `type(opts.shm_name) ~= "string"` already implies `opts.shm_name` can not be `false` or `nil`.

Reply:
Good catch @javierguerragiraldez - thanks!